src/toil/batchSystems/slurm.py

```diff
@@ -447,11 +500,15 @@ def prepareSbatch(self,
             if gpus:
                 # This block will add a gpu supported partition only if no partition is supplied by the user
                 sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
-                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i["gres"]], key=lambda x: x["priority"])
-                new_partition_name = lowest_priority_gpus[0][0]  # get name of lowest priority partition that supports gpus
                 if not any(option.startswith("--partition") for option in sbatch_line):
                     # no partition specified, so specify one
-                    sbatch_line.append(f"--partition={new_partition_name}")
+                    # try to get the name of the lowest priority gpu supported partition
+                    lowest_gpu_partition = self.boss.partitions.default_gpu_partition
+                    if lowest_gpu_partition is None:
+                        # no gpu partitions are available, raise an error
+                        required_gpus = Requirer(dict(accelerator=parse_accelerator(gpus)))
+                        raise InsufficientSystemResources(required_gpus, "accelerators", [], details=["Could not find a Slurm partition that supports GPUs!"])
+                    sbatch_line.append(f"--partition={lowest_gpu_partition}")
```

Comment:
If the lowest GPU partition is a PartitionInfo, this needs to actually get the name.

```suggestion
                    sbatch_line.append(f"--partition={lowest_gpu_partition.name}")
```